pinot-common/src/test/java/org/apache/pinot/common/utils/FALFInternerTest.java

```diff
@@ -0,0 +1,168 @@
+    // (1217 ms vs 2230 ms) With custom hash function, FALFInterner's speed is about the
+    // same as the Guava interner.
+//    for (int j = 0; j < 3; j++) {
+//      long time0 = System.currentTimeMillis();
+//      long totNHits = 0;
+//      for (int i = 0; i < 10000; i++) {
+//        totNHits += runInterning(allObjs, exactInterner, false);
+//      }
+//      long time1 = System.currentTimeMillis();
+//      System.out.println("Guava interner. totNHits = " + totNHits + ", time = " + (time1 - time0));
+//
+//      time0 = System.currentTimeMillis();
+//      totNHits = 0;
+//      for (int i = 0; i < 10000; i++) {
+//        totNHits += runInterning(allObjs, falfInterner, false);
+//      }
+//      time1 = System.currentTimeMillis();
+//      System.out.println("FALF interner. totNHits = " + totNHits + ", time = " + (time1 - time0));
+//
+//      time0 = System.currentTimeMillis();
+//      totNHits = 0;
+//      for (int i = 0; i < 10000; i++) {
+//        totNHits += runInterning(allObjs, falfInternerCustomHash, false);
+//      }
+//      time1 = System.currentTimeMillis();
+//      System.out.println("FALF interner Custom Hash. totNHits = " + totNHits + ", time = " + (time1 - time0));
+//    }
+  }
+
+  private int runInterning(String[] objs, Interner<String> interner, boolean performAssert) {
+    int nHits = 0;
+    for (String origObj : objs) {
+      String internedObj = interner.intern(origObj);
+      if (performAssert) {
+        Assert.assertEquals(origObj, internedObj);
+      }
+      if (origObj != internedObj) {
+        nHits++;
+      }
+    }
+    return nHits;
+  }
+
+  // Custom hash code implementation, that gives better distribution than standard hashCode()
+
+  private static final int C1 = 0xcc9e2d51;
+  private static final int C2 = 0x1b873593;
+
+  public static int hashCode(String s) {
+    int h1 = 0;
+
+    // step through value 2 chars at a time
+    for (int i = 1; i < s.length(); i += 2) {
+      int k1 = s.charAt(i - 1) | (s.charAt(i) << 16);
+      h1 = nextHashCode(k1, h1);
+    }
+
+    // deal with any remaining characters
+    if ((s.length() & 1) == 1) {
+      int k1 = s.charAt(s.length() - 1);
+      k1 = mixK1(k1);
+      h1 ^= k1;
+    }
+
+    return fmix(h1, s.length() * 2);
+  }
+
+  private static int nextHashCode(int value, int prevHashCode) {
+    int k1 = mixK1(value);
+    return mixH1(prevHashCode, k1);
+  }
+
+  private static int mixK1(int k1) {
+    k1 *= C1;
+    k1 = Integer.rotateLeft(k1, 15);
+    k1 *= C2;
+    return k1;
+  }
+
+  private static int mixH1(int h1, int k1) {
+    h1 ^= k1;
+    h1 = Integer.rotateLeft(h1, 13);
+    h1 = h1 * 5 + 0xe6546b64;
+    return h1;
+  }
+
+  private static int fmix(int h1, int len) {
+    // Force all bits to avalanche
+    h1 ^= len;
+    h1 ^= h1 >>> 16;
+    h1 *= 0x85ebca6b;
+    h1 ^= h1 >>> 13;
+    h1 *= 0xc2b2ae35;
+    h1 ^= h1 >>> 16;
+    return h1;
```

Comment:
I wonder why do we need to write all this complex code within a test class ?

Reply:
Shouldn't this be wrapped around in the form of a static helper class and then used in both test and actual code ?

Reply:
Ack. Addressed above.